node/src/components/consensus/highway_core/active_validator.rs

```diff
@@ -478,82 +483,268 @@ mod tests {
         }
     }
 
-    fn unwrap_single<T: Debug>(vec: Vec<T>) -> T {
-        let mut iter = vec.into_iter();
+    struct TestState {
+        rng: TestRng,
+        state: State<TestContext>,
+        instance_id: u64,
+        fd: FinalityDetector<TestContext>,
+        validators: HashMap<ValidatorIndex, ActiveValidator<TestContext>>,
+        timers: BTreeSet<(Timestamp, ValidatorIndex)>,
+    }
+
+    impl TestState {
+        fn new(
+            state: State<TestContext>,
+            rng: TestRng,
+            start_time: Timestamp,
+            instance_id: u64,
+            fd: FinalityDetector<TestContext>,
+            validators: Vec<ValidatorIndex>,
+        ) -> Self {
+            let mut timers = BTreeSet::new();
+            let earliest_round_start = state::round_id(start_time, state.params().init_round_exp())
+                + (1 << state.params().init_round_exp()).into();
+            let validator_map = validators
+                .into_iter()
+                .map(|vidx| {
+                    let secret = TestSecret(vidx.0);
+                    let (av, effects) = ActiveValidator::new(vidx, secret, start_time, &state);
+                    if let [Eff::ScheduleTimer(timestamp)] = *effects {
+                        // We start at time 410, with round length 16, so the first leader tick is
+                        // 416, and the first witness tick 426.
```

Comment:
That now depends on `state.params()`.

Reply:
That's true. Initially I didn't want to change too much and planned on doing that later on but I will fix it and push.

Reply:
Fixed in [`3e59e5b` (#602)](https://github.com/CasperLabs/casper-node/pull/602/commits/3e59e5be187011e0da8f7801704d291ba64967fc).